source/common/http/websocket/BUILD

```diff
@@ -21,9 +21,11 @@ envoy_cc_library(
         "//include/envoy/router:router_interface",
         "//include/envoy/upstream:cluster_manager_interface",
         "//source/common/common:enum_to_int",
-        "//source/common/filter:tcp_proxy_lib",
         "//source/common/http:codes_lib",
         "//source/common/http:header_map_lib",
         "//source/common/http/http1:codec_lib",
+        # TODO(mattklein123): Common code reaching into extensions is not right. Sort this out
+        # when the HTTP connection manager is moved.
+        "//source/extensions/filters/network/tcp_proxy",
```

Comment:
Arguably tcp_proxy (along with HCM) could be considered part of core Envoy vs. an extension.

Reply:
Yeah, I had the same thought.

Reply:
I also had the same thought, but here is my thinking on this: Even though I think it will be impossible to compile out tcp_proxy and hcm (primarily due to admin server), I still think there is value in colocating these things in the extensions directory, primarily because they are loaded/implemented as extensions. IMO it's better for code discoverability that all the extensions be together, along with configuration, potentially common extension code, etc. It's very reasonable to say that some extensions are locked and can't be compiled out.

The alternatives are:
1) Leave tcp_proxy and hcm where they are (I'm not a fan of this given code organization issues).
2) Make a "core_extensions" folder alongside "extensions" (or something like that) with the same folder structure to make it very clear that core extensions cannot be compiled out.

Thoughts?

Reply:
Filed https://github.com/envoyproxy/envoy/issues/2910 to officially look into this.